ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/HardFork/Combinator/Serialisation/Common.hs

```diff
@@ -210,6 +218,13 @@ isHardForkNodeToClientEnabled :: HardForkNodeToClientVersion xs -> Bool
 isHardForkNodeToClientEnabled HardForkNodeToClientEnabled {} = True
 isHardForkNodeToClientEnabled _                              = False
 
+
+-- absolutely horrible. should be HasLatestStableGenTxIdEra probably.
+-- especially nasty because it's a K () rather than an equivalent (but
+-- non-existent) 'Void :: (Type -> Type) -> Type'
+class HasBlessedGenTxIdEra (xs :: [Type]) where
+  blessedGenTxIdEra :: NS (K ()) xs
```

Comment:
Just brainstorming an alternative: Instead of having a typeclass for this, we could also "hardcode" this choice in the serialization logic (e.g. sth like: "use the first index when `xs` has length 1, otherwise, use the second index). The downside is that one can't customize this for a hypothetical different `HardForkBlock` that would need to customize this, but that seems unlikely.